Sources/Fluent/Pivot/PivotProtocol.swift

```diff
@@ -59,8 +59,11 @@ extension PivotProtocol where Self: Entity {
         let rightId = try right.assertExists()
 
         try makeQuery()
-            .filter(Left.foreignIdKey, leftId)
-            .filter(Right.foreignIdKey, rightId)
+            .filter(pivotLeftIdKey, leftId)
+            .filter(pivotRightIdKey, rightId)
             .delete()
     }
 }
+
+public var pivotLeftIdKey: String = "left_id"
```

Comment:
If you use the foreign key name of the type (like it is implemented now) you'll not be able to create a many-to-many relation between models of the same type.
Use case: I have a Contact model and I want to create a relation between them. Now I cannot use the PivotProtocol to do that.
Could we add a prefix or a postfix to the type foreign key, like `l_contact_id` and `r_contact_id` for my use case?